src/main/java/software/amazon/qldb/tutorial/qldb/QldbRevision.java

```diff
@@ -182,9 +182,8 @@ public int hashCode() {
     public void verifyRevisionHash() {
         // Certain internal-only system revisions only contain a hash which cannot be
         // further computed. However, these system hashes still participate to validate
-        // the journal block. User revisions will always contain values for all fields
-        // and can therefore have their hash computed.
-        if (blockAddress == null && metadata == null && data == null) {
+        // the journal block.
+        if (blockAddress == null || metadata == null || data == null) {
```

Comment:
Is it true? It has discrepancy with the words in [developer guide](https://docs.aws.amazon.com/qldb/latest/developerguide/journal-contents.html):

> In the revisions field, some revision objects might only contain a hash value and no other attributes. These are internal-only system revisions that don't contain user data. The hashes of these revisions are part of the journal's full hash chain, which is required for cryptographic verification.

Reply:
A deleted document will have a final revision that has metadata only https://docs.aws.amazon.com/qldb/latest/developerguide/working.history.html